src/P/List.hs

```diff
@@ -45,3 +46,11 @@ lastMaybe = listToMaybe . reverse
 -- | count the number of elements satisfying a predicate in a list
 count :: (a -> Bool) -> [a] -> Int
 count predicate = length . filter predicate
+
+toNonEmpty :: [a] -> Maybe (NonEmpty a)
+toNonEmpty l =
+  case l of
+    h : t ->
+      Just (h :| t)
+    _ ->
+      Nothing
```

Comment:
[this one](https://hackage.haskell.org/package/semigroups-0.18.1/docs/Data-List-NonEmpty.html#v:nonEmpty) no good?